index.js

```diff
@@ -13,83 +13,99 @@ token_utils._tokenCache = tokenCache; // Exposed for testing
 // This will fetch and cache the public key of the UAA used for this tenant.
 // This key can them be used to verify the JWT token presented so that the
 // details contained within the token can be trusted.  Such as the user, expiry and scopes.
-const getKey = (keyURL) => {
-    // URL for the token is <UAA_Server>/token_key
-    return new Promise((resolve, reject) => {
-        // Check the cache
-        if (oauthKeyCache[keyURL]) {
-            // Already have it.
-            resolve(oauthKeyCache[keyURL]);
-        } else {
-            // Fetch it and cache it for later
-            debug('Fetching key from:', keyURL);
-            rp.get(keyURL).then(body => {
-                debug('Fetched key');
-                const data = JSON.parse(body);
-                // Cache it
-                oauthKeyCache[keyURL] = data.value;
-                resolve(data.value);
-            }).catch(err => {
-                debug('Error reading token key from', keyURL, err);
-                reject(err);
-            });
-        }
-    });
+const getKey = (keyURL, tenantUuid) => {
+  // URL for the token is <UAA_Server>/token_key
+  return new Promise((resolve, reject) => {
+    // Check the cache
+    if (tenantUuid && oauthKeyCache[keyURL + tenantUuid]) {
```

Comment:
You may want to consider a delimiter between the URL and UUID, in case it's accidentally passed as an empty string.

Reply:
Good catch. Will add one.